Sources/StreamChat/Utils/EventBatcher.swift

```diff
@@ -59,19 +60,19 @@ final class Batcher<Item> {
             self.batchProcessingTimer = self.timerType.schedule(
                 timeInterval: self.period,
                 queue: self.queue,
-                onFire: self.process
+                onFire: { self.process() }
             )
         }
     }
     
-    func processImmediately() {
+    func processImmediately(completion: @escaping () -> Void) {
         timerType.schedule(timeInterval: 0, queue: queue) { [weak self] in
-            self?.process()
+            self?.process(completion: completion)
         }
     }
     
-    private func process() {
-        handler(currentBatch)
+    private func process(completion: (() -> Void)? = nil) {
```

Comment:
@polqf made `completion` optional here so we don't need to pass it when the timer fires